apis/v1alpha1/shared_types.go

```diff
@@ -80,77 +87,53 @@ type Port struct {
 type PortRange struct {
 	// Protocol is the network protocol (TCP, UDP, or SCTP) which traffic must
 	// match. If not specified, this field defaults to TCP.
+	// +kubebuilder:default=TCP
+	//
 	Protocol v1.Protocol `json:"protocol,omitempty"`
 
 	// Start defines a network port that is the start of a port range, the Start
 	// value must be less than End.
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=65535
+	//
 	Start int32 `json:"start"`
 
 	// End defines a network port that is the end of a port range, the End value
 	// must be greater than Start.
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=65535
+	//
 	End int32 `json:"end"`
 }
 
-// AdminNetworkPolicyPeer defines an in-cluster peer to allow traffic to/from.
-// Exactly one of the selector pointers must be set for a given peer. If a
-// consumer observes none of its fields are set, they must assume an unknown
-// option has been specified and fail closed.
+// AdminNetworkPolicyIngressPeer defines a peer to allow traffic to.
+//
+// Exactly one of the fields must be set for a given peer and this is enforced
+// by the validation rules on the CRD. If an implementation sees no fields are
+// set then it can infer that the deployed CRD is of an incompatible version
+// with an unknown field.  In that case it should fail closed.
+//
+// For "Allow" rules, "fail closed" means: "treat the rule as matching no
+// traffic". For "Deny" and "Pass" rules, "fail closed" means: "treat the rule
+// as a 'Deny all' rule".
+//
 // +kubebuilder:validation:MaxProperties=1
 // +kubebuilder:validation:MinProperties=1
-type AdminNetworkPolicyPeer struct {
-	// Namespaces defines a way to select a set of Namespaces.
+type AdminNetworkPolicyIngressPeer struct {
+	// Namespaces defines a way to select all pods within a set of Namespaces.
+	// Note that host-networked pods are not included in this type of peer.
+	//
 	// +optional
-	Namespaces *NamespacedPeer `json:"namespaces,omitempty"`
+	Namespaces *metav1.LabelSelector `json:"namespaces,omitempty"`
```

Comment:
The removal of `SameLabels`/`NotSameLabels` meant we no longer needed an additional struct here